lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts

```diff
@@ -0,0 +1,272 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getUser } from "../..";
+import RecipeUserId from "../../recipeUserId";
+import { SessionClaimValidator } from "../session";
+import { SessionClaim } from "../session/claims";
+import { JSONObject } from "../usermetadata";
+import { MFAClaimValue, MFARequirementList } from "./types";
+import Multitenancy from "../multitenancy";
+import MultiFactorAuthRecipe from "./recipe";
+import { DEFAULT_TENANT_ID } from "../multitenancy/constants";
+import { UserContext } from "../../types";
+
+/**
+ * We include "Class" in the class name, because it makes it easier to import the right thing (the instance) instead of this.
+ * */
+export class MultiFactorAuthClaimClass extends SessionClaim<MFAClaimValue> {
+    constructor(key?: string) {
+        super(key ?? "st-mfa");
+
+        this.validators = {
+            hasCompletedMFARequirementForAuth: (id?: string) => ({
```

Comment:
the input `id` should be renamed to `claimKey` cause otherwise it's confusing with factorId